include/mega/node.h

```diff
@@ -249,42 +249,42 @@ struct MEGA_API Node : public NodeCore, FileFingerprint
 #ifdef ENABLE_SYNC
 struct MEGA_API LocalNode : public File
 {
-    class Sync* sync;
+    class Sync* sync = nullptr;
 
     // parent linkage
-    LocalNode* parent;
+    LocalNode* parent = nullptr;
 
     // stored to rebuild tree after serialization => this must not be a pointer to parent->dbid
-    int32_t parent_dbid;
+    int32_t parent_dbid = 0;
 
     // children by name
     localnode_map children;
 
     // for botched filesystems with legacy secondary ("short") names
-    string *slocalname;
+    string* slocalname = nullptr;
     localnode_map schildren;
 
     // local filesystem node ID (inode...) for rename/move detection
-    handle fsid;
-    handlelocalnode_map::iterator fsid_it;
+    handle fsid = mega::UNDEF;
+    handlelocalnode_map::iterator fsid_it{};
 
     // related cloud node, if any
-    Node* node;
+    Node* node = nullptr;
 
     // related pending node creation or NULL
-    NewNode* newnode;
+    NewNode* newnode = nullptr;
 
     // FILENODE or FOLDERNODE
-    nodetype_t type;
+    nodetype_t type = TYPE_UNKNOWN;
 
     // detection of deleted filesystem records
-    int scanseqno;
+    int scanseqno = 0;
 
     // number of iterations since last seen
-    int notseen;
+    int notseen = 0;
 
     // global sync reference
-    handle syncid;
+    handle syncid = mega::UNDEF;
 
     struct
```

Comment:
Sadly, bit fields cannot be initialized at declaration (coming with c++20 though).